packages/dx-react-chart/src/plugins/tooltip.jsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {
+  Plugin,
+  Getter,
+  TemplateConnector,
+  Template,
+  TemplatePlaceholder,
+  withComponents,
+} from '@devexpress/dx-react-core';
+import { getParameters, processPointerMove } from '@devexpress/dx-chart-core';
+import { Target } from '../templates/tooltip/target';
+
+class RawTooltip extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      target: props.targetItem || props.defaultTargetItem,
```

Comment:
For *controlled/uncontrolled* approach `getDerivedStateFromProps` should also be defined.

Reply:
done